types/mempool/priority_nonce.go

```diff
@@ -16,6 +15,40 @@ var (
 	_ Iterator = (*PriorityNonceIterator)(nil)
 )
 
+type TxPriority struct {
+	// GetTxPriority returns the priority of the transaction. A priority must be
+	// comparable via CompareTxPriority.
+	GetTxPriority func(ctx context.Context, tx sdk.Tx) any
+	// CompareTxPriority compares two transaction priorities. The result should be
+	// 0 if a == b, -1 if a < b, and +1 if a > b.
+	CompareTxPriority func(a, b any) int
+}
+
+// NewDefaultTxPriority returns a TxPriority comparator using ctx.Priority as
+// the defining transaction priority.
+func NewDefaultTxPriority() TxPriority {
+	return TxPriority{
+		GetTxPriority: func(goCtx context.Context, tx sdk.Tx) any {
+			return sdk.UnwrapSDKContext(goCtx).Priority()
+		},
+		CompareTxPriority: func(a, b any) int {
+			switch {
+			case a == nil && b == nil:
```

Comment:
I'm wondering why there is nil handling when the priority getter (defined just above) cannot return nil.

Reply:
Tests failed without it. Recall, it's also used for weight/iteration and there is a case where it's set to nil:

```go
	nextPriorityNode := i.priorityNode.Next()
	if nextPriorityNode != nil {
		// ...
	} else {
		i.nextPriority = nil
	}
```